maps/Hydration/HDRUK/4.0.0/translation.jsonata

```diff
@@ -0,0 +1,390 @@
+(
+
+$isHidden := function($location){
+    $sum($map($hiddenFields, function($h){
+        $contains($location, $h) ? 1 : 0
+    })) >= 1 ? true : false
+};
+
+$allOptionalFields := [
+    "materialType",
+    "imageContrast",
+    "omics"
+];
+
+$neededFieldsMap := {
+    "Health and disease": ["materialType"],
+    "Treatments/Interventions": ["materialType"],
+    "Measurements/Tests": ["materialType"],
+    "Imaging types": ["imageContrast"],
+    "Imaging area of the body": ["imageContrast"],
+    "Omics": ["materialType","omics"],
+    "Socioeconomic": [],
+    "Lifestyle": [],
+    "Registry": ["materialType"],
+    "Environment and energy": [],
+    "Information and communication": [],
+    "Politics": [],
+    "": ["imageContrast","materialType","omics"]
+};
+
+$getNeededFields := function($dataTypes, $neededFieldsMap) {(
+    $list :=  $each($neededFieldsMap, function($v, $k) {
+        $k in dataTypes ? $v : []
+    });
+    $neededFields := $append([], $list).*
```

Comment:
Is the `.*` needed with `append[]`? Syntax is a bit hard to understand.  https://docs.jsonata.org/higher-order-functions#reduce might be better here?